File: go/chapter9/number_served.go
package main

import "fmt"

type Restaurant struct {
	name         string
	cuisineType  string
	numberServed int
}

func (res Restaurant) describeRestaurant() {
	fmt.Printf("Name: %s\n", res.name)
	fmt.Printf("Cuisine Type: %s\n", res.cuisineType)
}

func (res Restaurant) openRestaurant() {
	fmt.Printf("The %s Restaurant is now open!\n", res.name)
}

func (res *Restaurant) setNumberServed(num int) {
	res.numberServed = num
}

func (res *Restaurant) incrementNumberServed(num int) {
	res.numberServed += num
}

func main() {
	restaurant := Restaurant{name: "Sunny", cuisineType: "Classic"}

	fmt.Printf("The %s has served %d customers.\n", restaurant.name, restaurant.numberServed)
	restaurant.numberServed = 3
	fmt.Printf("The %s has served %d customers.\n", restaurant.name, restaurant.numberServed)
	restaurant.setNumberServed(10)
	fmt.Printf("The %s has served %d customers.\n", restaurant.name, restaurant.numberServed)
	restaurant.incrementNumberServed(5)
	fmt.Printf("The %s has served %d customers.\n", restaurant.name, restaurant.numberServed)
}

File: python/chapter9/number_served.py
class Restaurant:
    def __init__(self, restaurant_name: str, cuisine_type: str):
        self.restaurant_name = restaurant_name
        self.cuisine_type = cuisine_type
        self.number_served = 0

    def describe_restaurant(self):
        print(f"Restaurant Name: {self.restaurant_name}")
        print(f"Cuisine Type: {self.cuisine_type}")

    def open_restaurant(self):
        print(f"The {self.restaurant_name} Restaurant is now open!")

    def set_number_served(self, num: int):
        self.number_served = num

    def increment_number_served(self, num: int):
        self.number_served += num


restaurant = Restaurant("Sunny", "Classic")

print(restaurant.restaurant_name)
print(restaurant.cuisine_type)

restaurant.open_restaurant()
restaurant.describe_restaurant()


print(
    f"The {restaurant.restaurant_name} has served {restaurant.number_served} customers."
)
restaurant.number_served = 3
print(
    f"The {restaurant.restaurant_name} has served {restaurant.number_served} customers."
)

restaurant.set_number_served(10)
print(
    f"The {restaurant.restaurant_name} has served {restaurant.number_served} customers."
)
restaurant.increment_number_served(5)
print(
    f"The {restaurant.restaurant_name} has served {restaurant.number_served} customers."
)
